holmes/core/transformers/llm_summarize.py

```diff
@@ -0,0 +1,178 @@
+"""
+LLM Summarize Transformer for fast model summarization of large tool outputs.
+"""
+
+import logging
+from typing import Optional, ClassVar
+from pydantic import Field, PrivateAttr
+
+from .base import BaseTransformer, TransformerError
+from ..llm import DefaultLLM, LLM
+
+logger = logging.getLogger(__name__)
+
+
+class LLMSummarizeTransformer(BaseTransformer):
+    """
+    Transformer that uses a fast LLM model to summarize large tool outputs.
+
+    This transformer applies summarization when:
+    1. A fast model is available
+    2. The input length exceeds the configured threshold
+
+    Configuration options:
+    - input_threshold: Minimum input length to trigger summarization (default: 1000)
+    - prompt: Custom prompt template for summarization (optional)
+    - fast_model: Fast model name for summarization (e.g., "gpt-4o-mini")
+    - api_key: API key for the fast model (optional, uses default if not provided)
+    """
+
+    DEFAULT_PROMPT: ClassVar[str] = """Summarize this operational data focusing on:
+- What needs attention or immediate action
+- Group similar entries into a single line and description
+- Make sure to mention outliers, errors, and non-standard patterns
+- List normal/healthy patterns as aggregate descriptions
+- When listing problematic entries, also try to use aggregate descriptions when possible
+- When possible, mention exact keywords, IDs, or patterns so the user can filter/search the original data and drill down on the parts they care about (extraction over abstraction)"""
+
+    # Pydantic fields with validation
+    input_threshold: int = Field(
+        default=1000, ge=0, description="Minimum input length to trigger summarization"
+    )
+    prompt: Optional[str] = Field(
+        default=None,
+        min_length=1,
+        description="Custom prompt template for summarization",
+    )
+    fast_model: Optional[str] = Field(
+        default=None,
+        min_length=1,
+        description="Fast model name for summarization (e.g., 'gpt-4o-mini')",
+    )
+    global_fast_model: Optional[str] = Field(
+        default=None,
+        min_length=1,
+        description="Global fast model name fallback when fast_model is not set",
+    )
+    api_key: Optional[str] = Field(
+        default=None,
+        description="API key for the fast model (optional, uses default if not provided)",
+    )
+
+    # Private attribute for the LLM instance (not serialized)
+    _fast_llm: Optional[LLM] = PrivateAttr(default=None)
+
+    def model_post_init(self, __context) -> None:
+        """Initialize the fast LLM instance after model validation."""
+        import logging
```

Comment:
Please move import and logger definition out of the method to module level

Reply:
done